### PR TITLE
return error whrn app name was empty and remove useless code

### PR DESCRIPTION
**Pull request by upccup**

### CODE REVIEW

File: src/manager/framework/state/app.go [hunk 1]
```diff
@@ -73,6 +73,11 @@ type App struct {
 
 func NewApp(version *types.Version,
 	userEventChan chan *event.UserEvent) (*App, error) {
+	appID := fmt.Sprintf("%s-%s-%s", version.AppName, version.RunAs, mesos_connector.Instance().ClusterID)
+	existedApp, _ := persistentStore.GetApp(appID)
```
Comment (cmingxu) [inline src/manager/framework/state/app.go]: existing

File: src/manager/framework/state/app.go [hunk 2]
```diff
@@ -512,6 +517,10 @@ func validateAndFormatVersion(version *types.Version) error {
 		return errors.New("swan only support mesos docker containerization, no container found")
 	}
 
+	if version.AppName == "" {
```
Comment (cmingxu) [inline src/manager/framework/state/app.go]: regexp already includes check version.AppName empty

**Comment by cmingxu** [comment]
LGTM

File: src/manager/framework/state/app.go (hunk 2 above)
Comment (upccup) [inline src/manager/framework/state/app.go]: experiment display is out of coverage

**Pull request merged**